apps/full-site-editing/full-site-editing-plugin/full-site-editing/plugins/template-selector-sidebar/index.js

```diff
@@ -24,45 +26,83 @@ const TemplateSelectorSidebar = compose(
 	withDispatch( dispatch => ( {
 		setTemplateId: templateId =>
 			dispatch( 'core/editor' ).editPost( { meta: { _wp_template_id: templateId } } ),
+		resetBlocks: dispatch( 'core/block-editor' ).resetBlocks,
 	} ) ),
 	withSelect( select => {
 		const { canUser, getEntityRecord } = select( 'core' );
-		const templateId = get(
-			select( 'core/editor' ).getEditedPostAttribute( 'meta' ),
-			'_wp_template_id'
-		);
+		const { getBlocks, getEditedPostAttribute } = select( 'core/editor' );
+		const templateId = get( getEditedPostAttribute( 'meta' ), '_wp_template_id' );
+
 		return {
 			canUserUpdateSettings: canUser( 'update', 'settings' ),
+			contentBlocks: getBlocks(),
 			selectedTemplate: templateId && getEntityRecord( 'postType', 'wp_template', templateId ),
+			templateId,
 		};
 	} )
-)( ( { canUserUpdateSettings, setTemplateId, selectedTemplate } ) => {
-	if ( ! canUserUpdateSettings ) {
-		return null;
-	}
+)(
+	( {
+		canUserUpdateSettings,
+		contentBlocks,
+		resetBlocks,
+		setTemplateId,
+		selectedTemplate,
+		templateId,
+	} ) => {
+		if ( ! canUserUpdateSettings ) {
+			return null;
+		}
+
+		useEffect( () => {
+			const [ templateMerged, setTemplateMerged ] = useState( false );
+
+			if ( ! selectedTemplate || templateMerged ) {
+				return;
+			}
+
+			const templateBlocks = parse( get( selectedTemplate, [ 'content', 'raw' ] ) );
 
-	const onSelectTemplate = ( { id } ) => {
-		setTemplateId( parseInt( id, 10 ) );
-	};
+			const mergedBlocks = templateBlocks.map( ( { name, attributes } ) => {
+				if ( 'a8c/post-content' === name ) {
+					return createBlock(
+						'a8c/post-content', //for clarity
+						attributes,
+						contentBlocks.map( contentBlock =>
+							createBlock( contentBlock.name, contentBlock.attributes )
+						)
+					);
+				}
 
-	return (
-		<Fragment>
-			<PluginSidebarMoreMenuItem target="fse-template-sidebar" icon="layout">
-				{ __( 'Template' ) }
-			</PluginSidebarMoreMenuItem>
-			<PluginSidebar icon="layout" name="fse-template-sidebar" title={ __( 'Template' ) }>
-				<PanelBody>
-					{ __( 'Select a template' ) }
-					<PostAutocomplete
-						initialValue={ get( selectedTemplate, [ 'title', 'rendered' ] ) }
-						onSelectPost={ onSelectTemplate }
-						postType="wp_template"
-					/>
-				</PanelBody>
-			</PluginSidebar>
-		</Fragment>
-	);
-} );
+				return createBlock( name, attributes );
+			} );
+
+			resetBlocks( mergedBlocks );
+			setTemplateMerged( true );
+		}, [ templateId ] );
+
+		const onSelectTemplate = ( { id } ) => {
+			setTemplateId( parseInt( id, 10 ) );
+		};
+
+		return (
+			<Fragment>
```

Comment:
We can use the shorthand `<>` if you like (available since 16.2).

```suggestion
			<>
```